Model/Sync/CollectionsProducts/Services/CollectionsProductsService.php

```diff
@@ -0,0 +1,218 @@
+        parent::__construct($appEmulation, $resourceConnection);
+    }
+
+    /**
+     * @param integer $collectionsProductsSyncBatchSize
+     * @return array<string>
+     */
+    public function getCollectionsProductsToSync($collectionsProductsSyncBatchSize) {
+        $storeId = $this->coreConfig->getStoreId();
+        $connection = $this->resourceConnection->getConnection();
+        $categoryProductsQuery = $connection->select(
+        )->from(
+            [ $this->resourceConnection->getTableName($this::YOTPO_COLLECTIONS_PRODUCTS_SYNC_TABLE_NAME) ],
+            ['*']
+        )->where(
+            'magento_store_id = ?',
+            $storeId
+        )->where(
+            'is_synced_to_yotpo = ?',
+            false
+        )->limit(
+            $collectionsProductsSyncBatchSize
+        );
+
+        $collectionsProductsToSync = $connection->fetchAll($categoryProductsQuery);
+        return $collectionsProductsToSync;
+    }
+
+    /**
+     * @param string $categoryId
+     * @return array<string>
+     */
+    public function getCategoryProductsIdsFromSyncTable($categoryId) {
+        $connection = $this->resourceConnection->getConnection();
+        $categoryProductsQuery = $connection->select(
+        )->from(
+            ['entity' => $this->resourceConnection->getTableName($this::YOTPO_COLLECTIONS_PRODUCTS_SYNC_TABLE_NAME)],
+            ['magento_product_id']
+        )->where(
+            'magento_category_id = ?',
+            $categoryId
+        );
+
+        $categoryProductsIdsMap = $connection->fetchAssoc($categoryProductsQuery, 'magento_product_id');
+
+        $categoryProductsIds = [];
+        foreach ($categoryProductsIdsMap as $categoryProductsIdMap) {
+            $categoryProductsIds[] = $categoryProductsIdMap['magento_product_id'];
+        }
+
+        return $categoryProductsIds;
+    }
+
+    /**
+     * @param string $productId
+     * @return array<string>
+     */
+    public function getCategoryIdsFromSyncTableByProductId($productId) {
+        $connection = $this->resourceConnection->getConnection();
+        $categoryProductsQuery = $connection->select(
+        )->from(
+            ['entity' => $this->resourceConnection->getTableName($this::YOTPO_COLLECTIONS_PRODUCTS_SYNC_TABLE_NAME)],
+            ['magento_category_id']
+        )->where(
+            'magento_product_id = ?',
+            $productId
+        );
+
+        $categoryProductsIdsMap = $connection->fetchAssoc($categoryProductsQuery, 'magento_category_id');
+        return array_keys($categoryProductsIdsMap);
+    }
+
+    /**
+     * @param array $categoryProductsIds
+     * @param string $storeId
+     * @param string $categoryId
+     * @param boolean $isDeletedInMagento
+     * @return void
+     */
+    public function assignCategoryProductsForCollectionsProductsSync(array $categoryProductsIds, $storeId, $categoryId, $isDeletedInMagento = false)
```

Comment:
Let's change `categoryProductsIds` to `productIds` - the name of the method and the argument are super confusing (also relevant for `assignProductCategoriesForCollectionsProductsSync`).